pkgs/applications/networking/browsers/firefox/wrapper.nix

```diff
@@ -324,6 +324,8 @@ let
             --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH" \
             --suffix XDG_DATA_DIRS : '${gnome.adwaita-icon-theme}/share' \
             --set-default MOZ_ENABLE_WAYLAND 1 \
+            ${lib.optionalString (nativeMessagingHosts != []) "--run \"mkdir -p ~/.mozilla/native-messaging-hosts\""} \
+            ${lib.concatMapStringsSep " " (ext: "--run \"ln -sfLt ~/.mozilla/native-messaging-hosts ${ext}/lib/mozilla/native-messaging-hosts/*\"") nativeMessagingHosts} \
```

Comment:
Did not we do this before we had `MOZ_SYSTEM_DIR`? Why did it stop working?

Reply:
That requires a patch to firefox and though doesn't work with firefox-bin.

Reply:
Can we limit it to firefox-bin, then? Less stuff creating random junk in home directory, the better.

Reply:
Yeah, I've added a new argument weather our patch is included and based on that I choose the where to link the extensions to.